common/js/src/messaging/message-dispatcher.js

```diff
@@ -0,0 +1,93 @@
+/** @license
+ * Copyright 2016 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/** TODO @fileoverview */
+
+goog.provide('GoogleSmartCard.MessageDispatcher');
+
+goog.require('GoogleSmartCard.DebugDump');
+goog.require('GoogleSmartCard.Logging');
+goog.require('GoogleSmartCard.MessageChannelPinging.PingResponder');
+goog.require('GoogleSmartCard.MessageChannelPinging.Pinger');
+goog.require('GoogleSmartCard.OneTimeMessageChannel');
+goog.require('GoogleSmartCard.TypedMessage');
+goog.require('goog.asserts');
+goog.require('goog.events');
+goog.require('goog.log.Logger');
+goog.require('goog.messaging.AbstractChannel');
+goog.require('goog.structs.Map');
+
+goog.scope(function() {
+
+/** @const */
+var GSC = GoogleSmartCard;
+
+/**
+ * TODO: class description, @param?, @extends?
+ *
+ * @constructor
+ */
+GSC.MessageDispatcher = function() {
+  //MessageDispatcher.base(this, 'constructor');
+
+  /** TODO
+   * @type {!goog.log.Logger}
+   * @const
+   */
+  this.logger = GSC.Logging.getScopedLogger('OneTimeMessageDispatcher');
+
+  /**
+   * @type {goog.structs.Map.<string, !GSC.OneTimeMessageChannel>}
```

Comment:
Also, let's support having multiple channels for the same extensionId. This would allow tracking PortMessageChannel's without any further modifications.